pkg/apis/meta.go

```
@@ -57,33 +56,43 @@ type GrafanaResourceMetaAccessor interface {
 	SetFolder(uid string)
 	GetSlug() string
 	SetSlug(v string)
-	GetTitle() string
-	SetTitle(v string)
 	GetOriginInfo() (*ResourceOriginInfo, error)
 	SetOriginInfo(info *ResourceOriginInfo)
 	GetOriginName() string
 	GetOriginPath() string
 	GetOriginKey() string
 	GetOriginTimestamp() (*time.Time, error)
+
+	// Find a title in the object
+	// This will reflect the object and try to get:
+	//  * spec.title
+	//  * spec.name
+	//  * title
+	// and return an empty string if nothing was found
+	FindTitle(defaultTitle string) string
 }
 
 var _ GrafanaResourceMetaAccessor = (*grafanaResourceMetaAccessor)(nil)
 
 type grafanaResourceMetaAccessor struct {
-	obj runtime.Object
-}
-
-func MetaAccessor(obj runtime.Object) GrafanaResourceMetaAccessor {
-	return &grafanaResourceMetaAccessor{obj}
+	raw interface{} // the original object (it implements metav1.Object)
+	obj metav1.Object
 }
 
-func (m *grafanaResourceMetaAccessor) set(key string, val string) {
-	acc, err := meta.Accessor(m.obj)
+// Accessor takes an arbitrary object pointer and returns meta.Interface.
+// obj must be a pointer to an API type. An error is returned if the minimum
+// required fields are missing. Fields that are not required return the default
+// value and are a no-op if set.
+func MetaAccessor(raw interface{}) (GrafanaResourceMetaAccessor, error) {
```

this will now throw the error rather than silently failing at each request